src/openms/source/ANALYSIS/OPENSWATH/DIAScoring.cpp

```diff
@@ -399,11 +424,29 @@ namespace OpenMS
     {
       // create the theoretical distribution from the peptide weight
       CoarseIsotopePatternGenerator solver(dia_nr_isotopes_ + 1);
-      isotope_dist = solver.estimateFromPeptideWeight(std::fabs(product_mz * putative_fragment_charge));
+      isotope_dist = solver.estimateFromPeptideWeight(std::fabs((product_mz - C13C12_MASSDIFF_U * putative_fragment_charge) * putative_fragment_charge));
```

Comment:
Looks correct now but the difference in isotope intensities between new and old code will be super small

Reply:
yes I thought so too, but still, it is more correct.